### DISCUSSION TITLE
`rbenv version-file` command- printing `"${RBENV_ROOT}/version"` even when the file doesn't exist

### DISCUSSION THREAD

**Discussion by impostorsguides** [rbenv/rbenv#1510] [Q&A] [answered]
Hey y'all, I noticed that `rbenv version-file` [will default](https://github.com/rbenv/rbenv/blob/master/libexec/rbenv-version-file#L27) to printing `"${RBENV_ROOT}/version"` if the following two things are true:

 - no directory was specified as an argument, and
 - no `.ruby-version` was found for either `RBENV_DIR` or `PWD`.

There is a test which covers this behavior [here](https://github.com/rbenv/rbenv/blob/master/test/version-file.bats#L21), whose description states `"prints global file if no version files exist"`.

I get why `${RBENV_ROOT}/version` would be printed if it truly exists and performs the setting of the Ruby version, but printing it even if it doesn't exist is counter-intuitive to me.  The language `(set by /usr/local/rbenv/version)` seems to imply that the file actually exists, as does the language `detect the file that sets` in the "Summary" comments at the top of the command file.

Perhaps that interpretation is subjective; I'm happy to stipulate that.  But I might be confused if I had encountered this behavior in the wild and I hadn't yet read the source code.  From the way [this person phrased their question](https://github.com/rbenv/rbenv/issues/745), I might not be the only one who is similarly confused.

I'm guessing there's some context I'm missing which explains the behavior, just curious what that is.  Is it perhaps correct to say that the filepath printed by the command represents "where the version is expected to reside", rather than "where the version does reside"?  Thanks for your help.

**Comment by mislav** [comment, marked as answer]
> Is it perhaps correct to say that the filepath printed by the command represents "where the version is expected to reside", rather than "where the version does reside"?

That is correct. The command `rbenv version-file` not only prints locations of existing files, but also prints a [location of a file that, when created, would have dictated the version](https://github.com/rbenv/rbenv/pull/1203#pullrequestreview-340573721). I admit this is not clear and potentially confusing.

This is our current documentation for the command:
```
% rbenv version-file --help
Usage: rbenv version-file [<dir>]

Detect the file that sets the current rbenv version
```
This could definitely be better document. What do you think we could add to the documentation to clarify this?

**Reply by impostorsguides** [reply]
I know other commands have a `Summary` section, in addition to the `Usage` section.  I'd be happy to submit a PR adding such a section to this command as well, using the wording you used above.

**Reply by impostorsguides** [reply]
RE: the output of the command itself, assuming that brevity is a concern, what do you think about something like the following?

```
system (set by ${RBENV_ROOT}/version, if it exists)
```

This could obviously be applied to non-`system` output as well.

**Reply by mislav** [reply]
> RE: the output of the command itself, assuming that brevity is a concern, what do you think about something like the following?
> 
> ```
> system (set by ${RBENV_ROOT}/version, if it exists)
> ```

We cannot change the format of `rbenv version-file` output since that would break scripts.

**Reply by impostorsguides** [reply]
How about updating the `rbenv-version-file` command's docs to something like what you said above?

```
# Usage: rbenv version-file [<dir>]
# Summary: Detect the file that sets the current rbenv version.
#
# Prints locations of existing files.  If no file found, prints a location
# of a file that, when created, would have dictated the version.
```

**Reply by mislav** [reply]
@impostorsguides Sounds great! PR welcome.

**Reply by impostorsguides** [reply]
PR submitted [here](https://github.com/rbenv/rbenv/pull/1512) by my other GH account.
